src/components/TextField/TextField.tsx

```diff
@@ -204,6 +211,32 @@ export default class TextField extends React.PureComponent<Props, State> {
       </div>
     ) : null;
 
+    const characterCount = maxLength ? maxLength - value.length : value.length;
+    const characterCountLabel = intl.translate(
+      maxLength
+        ? 'Polaris.TextField.characterCountWithMaxLength'
+        : 'Polaris.TextField.characterCount',
+      {count: characterCount},
+    );
+
+    const characterCountClassName = classNames(
+      styles.CharacterCount,
+      multiline && styles.FlexEnd,
+      maxLength && characterCount === 0 && styles.MaxCharacters,
```

Comment:
If possible, it would be great to add code coverage for these conditions as well. (224 and 225)